src/edu/washington/escience/myria/daemon/MyriaDriverLauncher.java

```diff
@@ -212,9 +214,60 @@ private static int getDriverMemory(final Configuration conf) throws InjectionExc
   }
 
   /**
+   * Prints the expected configuration, compared to the actual configuration given.
+   * Used for logging command line arguments.
+   */
+  private static void logStartupMessage(Class<? extends Name<?>>[] classes, Injector inj) {
+    final StringBuilder sb = new StringBuilder("MyriaDriverLauncher configuration:\n");
+    for (int i = 0; i < classes.length; i++) {
+      final Class<? extends Name<?>> c = classes[i];
+      final NamedParameter annotation = c.getAnnotation(NamedParameter.class);
+
+      final String fullName = c.getName();
+      final String simpleName = c.getSimpleName();
+      final String shortName = annotation.short_name();
+      final String doc = annotation.doc();
+      final String defaultVal;
+      {
+        if (!annotation.default_value().equals(NamedParameter.REEF_UNINITIALIZED_VALUE))
+          defaultVal = annotation.default_value();
+        else if (!annotation.default_class().equals(Void.class))
+          defaultVal = annotation.default_class().getSimpleName();
+        else if (annotation.default_values().length > 0)
+          defaultVal = Arrays.toString(annotation.default_values());
+        else if (annotation.default_classes().length > 0) {
+          Class<?>[] default_classes = annotation.default_classes();
+          String[] sss = new String[default_classes.length];
+          for (int i1 = 0; i1 < default_classes.length; i1++) {
+            sss[i1] = default_classes[i1].getSimpleName();
+          }
+          defaultVal = Arrays.toString(sss);
+        } else defaultVal = "";
+      }
+
+      if (i > 0) sb.append('\n');
```

Comment:
You could avoid using an array index here by using Guava's `Joiner` instead of `StringBuilder`.